File: hw/old/hwhash1.cc
#include <iostream>
#include <word>
using namespace std;

int main() {
  ifstream words("dict.dat");
  string word;
  HashMapLinearChaining dict1(50000); // initial size
  while (!words.feof()) {
    words >> word;
    dict1.add(word); // add the word to the dictionary
  }
  dict1.hist(); // display histogram
  ifstream spell("words.dat");
  string s;
  while (!spell.eof()) {
    spell >> s;
    if (dict1.contains(s))
      cout  << s << " found!\n";
    else
      cout << s << " not found!\n";
  }
}
  

File: old/session10_Hashing/hwhash1.java
import java.io.*;

public class hwhash1 {
	public static void main(String[] args) throw Exception {
		FileReader fr = new FileReader("dict.dat");
		Scanner s = new Scanner(fr);
		String word;
		HashMapLinearChaining dict1 =
			new HashMapLinearChaining(50000); // initial size
    while (s.hasNext()) {
			word = s.next();
      dict1.add(word); // add the word to the dictionary
		}
		dict1.hist(); // display histogram
		s.close();
		//		fr.close();
		
		Scanner spell = new Scanner(new FileReader("words.dat"));
		
		while (spell.hasNext()) {
			word = spell.next();
			if (dict1.contains(word))
				System.out.println(word + " found!");
			else
				System.out.println(word + " not found!");
		}
	}
}
  
